canonicalization.md

```diff
@@ -0,0 +1,19 @@
+# Canonicalization
```

Comment:
This feels a bit weird lacking context - could we not wrap it into the document talking about the json types?